R/PipeOpDateFeatures.R

```diff
@@ -0,0 +1,226 @@
+#' functions from the `fastai` library. If operation on only particular `POSIXct` columns is
+#' requested, use the `affect_columns` parameter inherited from [`PipeOpTaskPreprocSimple`].
+#'
+#' @section Construction:
+#' ```
+#' PipeOpDateFeatures$new(id = "datefeatures", param_vals = list())
+#' ```
+#' * `id` :: `character(1)`\cr
+#'   Identifier of resulting object, default `"datefeatures"`.
+#' * `param_vals` :: named `list`\cr
+#'   List of hyperparameter settings, overwriting the hyperparameter settings that would otherwise
+#'   be set during construction. Default `list()`.
+#'
+#' @section Input and Output Channels:
+#' Input and output channels are inherited from [`PipeOpTaskPreprocSimple`].
+#'
+#' The output is the input [`Task`][mlr3::Task] with date-related features computed and added to the
+#' feature set of the output task and the `POSIXct` columns of the data removed from the
+#' feature set (depending on the value of `keep_date_var`).
+#'
+#' @section State:
+#' The `$state` is a named `list` with the `$state` elements inherited from
+#' [`PipeOpTaskPreprocSimple`].
+#'
+#' @section Parameters:
+#' The parameters are the parameters inherited from [`PipeOpTaskPreprocSimple`], as well as:
+#' * `keep_date_var` :: `logical(1)`\cr
+#'   Should the `POSIXct` columns be kept as features? Default FALSE.
+#' * `cyclic` :: `logical(1)`\cr
+#'   Should cyclic features be computed? See Internals? Default FALSE.
+#' * `year` :: `logical(1)`\cr
+#'   Should the year be extracted as a feature? Default TRUE.
+#' * `month` :: `logical(1)`\cr
+#'   Should the month be extracted as a feature? Default TRUE.
+#' * `week_of_year` :: `logical(1)`\cr
+#'   Should the week of the year be extracted as a feature? Default TRUE.
+#' * `day_of_year` :: `logical(1)`\cr
+#'   Should the day of the year be extracted as a feature? Default TRUE.
+#' * `day_of_month` :: `logical(1)`\cr
+#'   Should the day of the month be extracted as a feature? Default TRUE.
+#' * `day_of_week` :: `logical(1)`\cr
+#'   Should the day of the week be extracted as a feature? Default TRUE.
+#' * `hour` :: `logical(1)`\cr
+#'   Should the hour be extracted as a feature? Default TRUE.
+#' * `minute` :: `logical(1)`\cr
+#'   Should the minute be extracted as a feature? Default TRUE.
+#' * `second` :: `logical(1)`\cr
+#'   Should the second be extracted as a feature? Default TRUE.
+#' * `is_day` :: `logical(1)`\cr
+#'   Should a feature be extracted indicating whether it is day time (06:00am - 08:00pm)?
+#'   Default TRUE.
+#'
+#' @section Internals:
+#' If `cyclic = TRUE`, cyclic features are computed for the features `"month"`, `"week_of_year"`,
+#' `"day_of_year"`, `"day_of_month"`, `"day_of_week"`, `"hour"`, `"minute"` and `"second"`. This
+#' means that for each feature, `x`, two additional features are computed, namely the sine and
+#' cosine transformation of `2 * pi * x / max_x + 1`. This is useful to respect the cyclical nature
```

Comment:
I think the documentation should be clearer here:

1. what is `max_x`?
2. Why the `+ 1`?